stone/ir/data_types.py

```diff
@@ -1933,6 +1933,32 @@ def unwrap_aliases(data_type):
         data_type = data_type.data_type
     return data_type, unwrapped_alias
 
+def resolve_aliases(alias):
+    """
+    Resolve all chained / nested Aliases - this will recursively point
+    nested aliases to their resolved data type (first non-alias in the chain).
+
+    Note: This differs from unwrap_alias which simply identifies/returns
+    the resolved data type.
+
+    Args:
+        alias (Alias): The target Alias to resolve.
+    Return:
+        Alias: The resolved parent object.
+    """
+    def _resolve_data_type_chain(data_type):
+        if not is_alias(data_type):
+            return data_type
+
+        resolved = _resolve_data_type_chain(data_type.data_type)
+        data_type.data_type = resolved
+
+        return resolved
+
+    if hasattr(alias, 'data_type'):
```

Comment:
Hmm. I might miss something here. But what's the point of this checking? If `alias` is of type `Alias`, then `data_type` should always present since it's set in `__init__` (https://github.com/dropbox/stone/blob/6e8744a735eb9b68bce5730f5f2ea80cca5d5081/stone/ir/data_types.py#L1816).

I also don't see the point of wrapping this in an internal function. The original implementation looks good to me.

Reply:
Yeah I made a note to remove that check :)

I changed the implementation since it seemed a bit weird/arbitrary to expect a developer to pass in the alias's `data_type` attribute e.g. `resolve_aliases(alias.data_type)` as opposed to just passing in the alias e.g. `resolve_aliases(alias)` and handling it internally.